setup.sh

```diff
@@ -238,17 +249,31 @@ installDocker() {
   do
     echo
     writeBold "[❓] Do you want to install Docker now? (y/N)"
+    if [ $EUID != 0 ]; then
+      write "    Installing Docker needs root privileges."
+      write "    Since you are not running this script with root privileges"
+      write "    you are likely to be prompted for your sudo password."
+    fi
     echo -n "> "
     read installDocker trash
-    echo
     case $installDocker in
       [yY])
         if [ -z $DL_BIN ]; then
           failDL
         else
           writeBold "[ℹ] Installing Docker..."
-          $DL_BIN $DL_OPTS https://get.docker.com/ | sh
-          writeBold "$GREEN" "[✔] Docker successfully installed."
+          echo
+          sudo $DL_BIN $DL_OPTS https://get.docker.com/ | sudo sh
+          if [ $? -eq 0 ]; then
+            writeBold "$GREEN" "[✔] Docker successfully installed."
```

Comment:
Can we add a new step after this one, asking the user if (s)he wants to be added to the `docker` group?

Reply:
We considered this option and it resulted to be controversial. It basically boils down to the fact that, even if we add the user to the group, this only takes effect on the next login. Not very handy.

Reply:
Not a change request:

I don't see it as controversial, just not serving our immediate purpose with this script.
Docker and compose are actually the only requirements to run Kuzzle. All the rest is configuration and setting a compose stack up.

The more we discuss about it, the more I believe installing some service component on behalf of the user is actually causing more problems than help.

If docker or compose is missing, I would stop the script there, ask the user to install them, referring him to docker docs, and ask him to run it again once done.

Reply:
Would like to have @gandbox 's take here.

Reply:
@benoitvidis I don't know. I think the current behavior (i.e. installing Docker and docker-compose on behalf of the user *only* if the user agrees with it) is fairly good.
I think we should come to an agreement about what's the purpose of the script. From my point of view, it's all about smoothing the process of having a running instance of Kuzzle on your machine, be it your laptop or any kind of instance on a cloud hosting service. It's just that we don't want the user that wants to check Kuzzle out to be frustrated saying "crap, another complicated thing I have to spend hours just to make a POC". Quite the opposite. I'd like to achieve the WOW effect, when the thing _just works_, which is a very ambitious goal.